unified-codes/libs/data/src/lib/types/Entity.ts

```diff
@@ -83,6 +104,34 @@ export class Entity implements IEntity {
   matchesProperty(property: Property) {
     return this.properties?.includes(property);
   }
+
+  getParentProperties() {
+    let { parent } = this;
+    do {
+      if (parent?.length) {
+        const [parentEntity] = parent;
+        if (parentEntity.properties) return parentEntity.properties;
+        parent = parentEntity.parent;
+      } else {
+        parent = undefined;
+      }
+    } while (!!parent);
+  }
+
+  updatePropertiesFromParent() {
+    productPropertyTypes.forEach((type) => {
+      const parentProperty = this.getParentProperty(type);
+      const childProperty = this.getProperty(type);
+
+      if (parentProperty && !childProperty) {
+        if (!this.properties) {
+          this._properties = [new Property(parentProperty)];
+        } else {
+          this._properties.push(new Property(parentProperty));
+        }
+      }
+    });
+  }
```

Comment:
A little confused by this also... I think the niceness of the recursive structure is that we can traverse the tree? Since you've exposed `parent` as a getter, aren't these methods redundant (e.g. we don't have `getChildProperties`, as can just do `child.properties`)?

Reply:
Also double confused about `updatePropertiesFromParent`... what's the need to duplicate the properties on the child? As long as our schema is documented clearly, shouldn't the user know where the properties they want are located in the hierarchy?